src/main/java/com/box/sdk/BoxSignRequest.java

```diff
@@ -0,0 +1,812 @@
+package com.box.sdk;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+
+/**
+ * Represents a sign request used in Box Sign.
+ * Sign requests are used to prepare documents for signing and send them to signers.
+ *
+ * @see <a href="https://docs.box.com/reference#sign-request-object">Box sign request</a>
+ *
+ * <p>Unless otherwise noted, the methods in this class can throw an unchecked {@link BoxAPIException} (unchecked
+ * meaning that the compiler won't force you to handle it) if an error occurs. If you wish to implement custom error
+ * handling for errors related to the Box REST API, you should capture this exception explicitly.</p>
+ */
+@BoxResourceType("sign_request")
+public class BoxSignRequest extends BoxResource {
+
+    /**
+     * The URL template used for operation with sign requests.
+     */
+    public static final URLTemplate SIGN_REQUESTS_URL_TEMPLATE = new URLTemplate("sign_requests");
+
+    /**
+     * The URL template used for operation with sign request with given ID.
+     */
+    public static final URLTemplate SIGN_REQUEST_URL_TEMPLATE = new URLTemplate("sign_requests/%s");
+
+    /**
+     * The URL template used to cancel existing sign request.
+     */
+    public static final URLTemplate SIGN_REQUEST_CANCEL_URL_TEMPLATE = new URLTemplate("sign_requests/%s/cancel");
+
+    /**
+     * The URL template used to resend existing sign request.
+     */
+    public static final URLTemplate SIGN_REQUEST_RESEND_URL_TEMPLATE = new URLTemplate("sign_requests/%s/resend");
+
+    /**
+     * The default limit of entries per response.
+     */
+    private static final int DEFAULT_LIMIT = 100;
+
+    /**
+     * Constructs a BoxResource for a resource with a given ID.
+     *
+     * @param api the API connection to be used by the resource.
+     * @param id  the ID of the resource.
+     */
```

Comment:
Can we make this comment specific to Sign Request Resources? It wasn't clear to me what was the "id of the resource" when I was first reading it